core/unix/signal.c

```diff
@@ -815,6 +815,12 @@ create_clone_record(dcontext_t *dcontext, reg_t *app_thread_xsp)
             record->app_clone_args = app_clone_args;
         } else {
 #endif
+            /* TODO i#5365: If the record is created using this SP,
+             * get_clone_record() crashes with SIGSEGV when the record pointer
+             * is accessed in ASSERT(dstack_base == record->dstack). This does
+             * not happen when dr_clone_args->stack + dr_clone_args->stack_size
+             * (above) is used.
```

Comment:
I'm not sure I understand "created using this SP": are you talking about just the app_thread_xsp field?  But how would that sub-field affect a de-reference of the record pointer?

This else() is for non-clone3.  So this PR fails for non-clone3?  I suspect our linux.clone test uses the older clone syscalls: does that pass?

Reply:
Comment deleted as it no longer applies. Solution was to page align correctly in `get_clone_record()` and `client_thread_run()`.